yt/loaders.py

```diff
@@ -1561,3 +1561,107 @@ def del_callback(self):
     ds.dismount = types.MethodType(del_callback, ds)
 
     return ds
+
+
+def load_hdf5_file(
+    fn: Union[str, "os.PathLike[str]"],
+    root_node: Optional[str] = "/",
+    fields: Optional[List[str]] = None,
+    bbox: np.ndarray = None,
+    nchunks: int = 0,
```

Comment:
does nchunks=0 have meaning beyond being a flag to trigger the auto-chunking? I'm inclined to change this to 

```suggestion
    nchunks: Optional[int] = None,
```
and change the `nchunks` logic below to match. But I don't feel strongly about it.